lib/hashly.js

```diff
@@ -143,31 +146,48 @@ var processEntry = function (fullPath, baseDir, targetDir, data) {
         return;
     }
 
-    // Apply filters to exclude files if specified
-    if (_options.shouldBeExcluded) {
-        if (_options.shouldBeExcluded(fullPath)) {
-            return;
-        }
+    // Ignore map files, we check each css and js file for accompaning map file
+    if (ext === ".map") {
+        return;
     }
 
     try {
         if (_options.isPassthrough && _options.isPassthrough(fullPath)) {
-            
             // Passthrough means to simply copy file to the destination path, without labeling it with a hashcode or 
             // including in the manifest.
             var relativePath = path.relative(baseDir, fullPath);
             var targetPath = path.join(targetDir, relativePath);
             fsutil.copySync(fullPath, targetPath);
             return null;
-        }
-        else {
-            
-            var entry = createAndAugmentManifestEntry(fullPath, baseDir, targetDir, data);
+        } else {
+            // Improve map files to have the hashed filepath in their source (instead of unhashed filepath) - useful for debugging
+            // Note: assuming map files are in the same dir as js/css file and they end with .map
+            var mapFullPath = fullPath + ".map";
+            var transformedText = null,
+                hashCode = null;
+            if (_options.processMap && fsutil.existsSync(mapFullPath)) {
+                var mapEntry = createAndAugmentManifestEntry(mapFullPath, baseDir, targetDir, data);
+                fsutil.copySync(mapFullPath, mapEntry.hashedPathPhysical);
+                if (_options.logger) {
+                    _options.logger(mapFullPath + " > " + mapEntry.hashedPathPhysical);
+                }
+                // The sourcemap filename has changed, so update the minified js/css file and replace unhashed filename with hashed one
+                // This is done to support zero downtime deployment when two different releases are out at once

```

Comment:
This comment seems strangely general to be applied to this line- the whole point of hashly is to support zero-downtime deployments.